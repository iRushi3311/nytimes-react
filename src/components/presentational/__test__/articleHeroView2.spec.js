import React from 'react';
import renderer from 'react-test-renderer';
import ArticleHero2 from '../article-hero-view-2.js';
import Article from "../../../models/article";
import {MemoryRouter} from 'react-router-dom';

describe('>>>C O M P O N E N T S --- snapshot test for ArticleHero2', () => {

    let props,
        componentTree;

    beforeEach(() => {
        const src = {
            _id: 12345,
            web_url: 'google.com',
            headline:{
                main: 'So and so guy did bla bla bla'
            },
            snippet: 'body text body text body text body text body text body text ',
            byline:{
                original: 'By RPatel'
            },
            multimedia: [
                {
                    type: 'type-1',
                    subtype: 'subtype-X',
                    url: 'img_X',
                    height: '75px',
                    width: '75px'
                },
                {
                    type: 'type-3',
                    subtype: 'thumbnail',
                    url: 'img_Z',
                    height: '75px',
                    width: '75px'
                },
            ],
            pub_date: '2017-08-17T09:00:01+0000'
        };
        const articleObj = new Article(src);

        props = {
            article: articleObj
        };

        componentTree = renderer.create(
            <MemoryRouter>
                <ArticleHero2 {...props}/>
            </MemoryRouter>
        ).toJSON();
    });

    it('+++ ArticleHero2 match with snapshot ', () => {
        expect(componentTree).toMatchSnapshot();
    });

});