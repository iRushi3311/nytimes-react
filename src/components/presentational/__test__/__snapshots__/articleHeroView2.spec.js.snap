// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>C O M P O N E N T S --- snapshot test for ArticleHero2 +++ ArticleHero2 match with snapshot  1`] = `
<a
  href="/article/12345"
  onClick={[Function]}
>
  <div
    className="hero-contianer-2 article-container"
  >
    <h4
      className="hero-2-title-n2"
    >
       
      So and so guy did bla bla bla
       
    </h4>
    <h5
      style={
        Object {
          "marginBottom": "0px",
        }
      }
    >
      <span
        className="author-info"
      >
        By RPatel
      </span>
        
      <span
        className="published-time"
      >
        August 17  - 05:00 AM
      </span>
    </h5>
  </div>
</a>
`;
